llvm/lib/Target/AMDGPU/AMDGPUInstructionSelector.cpp

```diff
@@ -469,10 +469,14 @@ bool AMDGPUInstructionSelector::selectG_ADD_SUB(MachineInstr &I) const {
   const TargetRegisterClass &HalfRC
     = IsSALU ? AMDGPU::SReg_32RegClass : AMDGPU::VGPR_32RegClass;
 
-  MachineOperand Lo1(getSubOperand64(I.getOperand(1), HalfRC, AMDGPU::sub0));
-  MachineOperand Lo2(getSubOperand64(I.getOperand(2), HalfRC, AMDGPU::sub0));
-  MachineOperand Hi1(getSubOperand64(I.getOperand(1), HalfRC, AMDGPU::sub1));
-  MachineOperand Hi2(getSubOperand64(I.getOperand(2), HalfRC, AMDGPU::sub1));
+  const MachineOperand &Lo1(
+      getSubOperand64(I.getOperand(1), HalfRC, AMDGPU::sub0));
```

Comment:
Can't do this, this is a reference to an immediately dead temporary

Reply:
Removed dangling refs.